src/usr/local/www/diag_dns.php

```diff
@@ -84,12 +84,33 @@
 }
 
 function resolve_host_addresses($host) {	
-	$host_esc = escapeshellarg($host);
 	$resolved = array();
-	exec("/usr/bin/drill {$host_esc} A | /usr/bin/grep {$host_esc} | grep '\tA\t\|\tCNAME\t' | /usr/bin/grep -v ';' | /usr/bin/awk '{ print $5 }'", $resolved);
-	$resolved_v6 = array();
-	exec("/usr/bin/drill {$host_esc} AAAA | /usr/bin/grep {$host_esc} | grep '\tAAAA\t\|\tCNAME\t' | /usr/bin/grep -v ';' | /usr/bin/awk '{ print $5 }'", $resolved_v6);
-	return array_unique(array_merge($resolved, $resolved_v6));
+
+	$errreporting = error_reporting();
+	error_reporting($errreporting & ~E_WARNING);// dns_get_record throws a warning if nothing is resolved..
+
+	$dnsresult_cname = dns_get_record($host, DNS_CNAME);
+	$dnsresult_v4 = dns_get_record($host, DNS_A);
+	$dnsresult_v6 = dns_get_record($host, DNS_AAAA);	
```

Comment:
@PiBa-NL you mean previously cached by the configured DNS server? PHP issues new DNS queries every time dns_get_record is run. So yeah you might end up with a reply from cache from the configured DNS server, but I don't see how that's any diff regardless of whether it's DNS_ANY or DNS_A or DNS_AAAA.